pygeos/test/test_set_operations.py

```diff
@@ -41,6 +50,35 @@ def test_set_operation_array(a, func):
     assert isinstance(actual[0], Geometry)
 
 
+@pytest.mark.skipif(pygeos.geos_version >= (3, 9, 0), reason="GEOS >= 3.9")
+@pytest.mark.parametrize("func", SET_OPERATIONS)
+@pytest.mark.parametrize("grid_size", [0, 1])
+def test_set_operations_prec_not_supported(func, grid_size):
+    with pytest.raises(
+        UnsupportedGEOSOperation, match="grid_size parameter requires GEOS >= 3.9.0"
+    ):
+        func(point, point, grid_size)
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 9, 0), reason="GEOS < 3.9")
+@pytest.mark.parametrize("func", SET_OPERATIONS)
+def test_set_operation_prec_nonscalar_grid_size(func):
+    with pytest.raises(
+        ValueError, match="grid_size parameter only accepts scalar values"
+    ):
+        func(point, point, grid_size=[1])
+
+
+@pytest.mark.skipif(pygeos.geos_version < (3, 9, 0), reason="GEOS < 3.9")
+@pytest.mark.parametrize("a", all_single_types)
+@pytest.mark.parametrize("func", SET_OPERATIONS)
+@pytest.mark.parametrize("grid_size", [0])
+def test_set_operation_prec_array(a, func, grid_size):
+    actual = func([a, a], point, grid_size=grid_size)
+    assert actual.shape == (2,)
+    assert isinstance(actual[0], Geometry)
```

Comment:
Maybe also assert that the precision of the result is equal to grid size? (so we actually are sure we properly passed the grid size param, as I suppose this test would pass as well at the moment if the implementation had a mistake there)

(Eg for union_all below you have a few cases with actual input vs expected based on grid_size, which ensures things work as expected)

Reply:
Unfortunately, GEOS currently returns 0 for the precision of the outputs when `grid_size` is provided as a parameter here.

I added a test to compare the results to the output of the operation against geometries that have been set to the same `grid_size` in advance via `set_precision`.

Reply:
> Unfortunately, GEOS currently returns 0 for the precision of the outputs when grid_size is provided as a parameter here.

Ah, that's interesting to know (it's probably not "unfortunately", but intentionally done that way in GEOS?). So that is then still a difference between doing a `set_precision` first vs specifying the `grid_size` keyword. When setting the precision, that gets propagated to the output, while passing the keyword, the grid size is only used "within" the operation, but not set on the output.

Reply:
Added clarifying note to docstring.  Not sure if this is intentional in GEOS.

Reply:
Just checked on the geos chat channel, and Martin Davies confirmed that it is intentional

Reply:
Quoting him "The idea is to give more flexibility in using precision for specific operations, without affecting the underlying geometry precision model"